File: Module 3/Mastering Swift 2_Code/B05115_13_codes/Projects/ObjectiveCProject/ObjectiveCProject/Messages.m
//
//  Messages.m
//  ObjectiveCProject
//
//  Created by Jon Hoffman on 2/12/15.
//  Copyright (c) 2015 Jon Hoffman. All rights reserved.
//

#import "Messages.h"

@implementation Messages

NSMutableArray *theMessages;

-(id)init {
    if ( self = [super init] ) {
        theMessages = [[NSMutableArray alloc] init];
        [theMessages addObject:@"You should learn from your mistakes"];
        [theMessages addObject:@"It is in the now that we must live"];
        [theMessages addObject:@"The greatest risk is not taking one"];
        [theMessages addObject:@"You will be a Swift programmer"];
    }
    
    return self;
}

-(NSString *)getMessage {
    int num = arc4random() % theMessages.count;

    return theMessages[num];
}
@end

File: Module 3/Mastering Swift 2_Code/B05115_13_codes/Projects/SwiftProject/SwiftProject/Messages.swift
//
//  Messages.swift
//  SwiftProject
//
//  Created by Jon Hoffman on 2/13/15.
//  Copyright (c) 2015 Jon Hoffman. All rights reserved.
//

import Foundation

class Messages: NSObject {

    private var messages = [String]()
    
    override init() {
        messages.append("You should learn from your mistakes")
        messages.append("It is in the now that we must live")
        messages.append("The greatest risk is not taking one")
        messages.append("You will be a Swift programmer")
    }
    
    func getMessage() -> String {
        let num = Int(arc4random()) % messages.count
        return messages[num]
    }
}
